### PR TITLE
Update v1 in players.js to reflect players_info_expanded_all

### PR DESCRIPTION
**Pull request by ifhbiff**

### CODE REVIEW

**Comment by unknownunknowns1000** [comment]
How does players_info_expanded_all compare to players and current_players?

File: lib/db/players.js [hunk 1]
```diff
@@ -7,57 +7,20 @@ const {
...
-		soul,
-		fate,
-		peanut_allergy,
-		armor,
-		bat,
-		ritual,
-		coffee,
-		blood
-		FROM data.players WHERE deceased AND valid_until IS NULL`
+    `SELECT *
```
Comment (unknownunknowns1000) [inline lib/db/players.js:10]: 👍

File: lib/db/players.js [hunk 2]
```diff
@@ -7,57 +7,20 @@ const {
...
+		FROM data.players_info_expanded_all WHERE deceased AND valid_until IS NULL
+		ORDER BY player_name`
   );
 }
 
 // Get the first record with each player id, sorted by time (so the first is the most recent)
```
Comment (unknownunknowns1000) [inline lib/db/players.js]:
```suggestion
```
This comment is a lie

File: lib/db/players.js [hunk 3]
```diff
@@ -77,7 +40,8 @@ async function playerInfoCurrent(playerIds) {
     `SELECT * 
 		FROM data.players_info_expanded_all 
 		WHERE player_id=ANY($1)
-		AND valid_until is null`,
+		AND valid_until is null
```
Comment (unknownunknowns1000) [inline lib/db/players.js]:
```suggestion
		AND valid_until is NULL
```

File: lib/db/players.js [hunk 4]
```diff
@@ -87,41 +51,34 @@ async function playerInfoAll(playerIds) {
...
-		join data.teams t on t.team_id = tr.team_id
-		where p.valid_until is null
-		and tr.valid_until is null
-		and t.valid_until is null
-		and p.deceased = false
-		and pm.valid_until is null
-		order by valid_from`
+    `SELECT * 
+		FROM data.players_info_expanded_all 
+		WHERE valid_until IS NULL AND modifications IS NOT null order by player_name`
```
Comment (unknownunknowns1000) [inline lib/db/players.js]:
```suggestion
		WHERE valid_until IS NULL AND modifications IS NOT null ORDER BY player_name`
```

File: lib/db/players.js [hunk 5]
```diff
@@ -87,41 +51,34 @@ async function playerInfoAll(playerIds) {
...
-    : 'data.players_current';
-
   return await all(
-    `SELECT player_id, player_name, gameday_from, season_from, deceased, team_id, team, position_id, position_type
-		FROM ${view}
+    `SELECT *
+		FROM data.players_info_expanded_all 
+		WHERE valid_until IS NULL
+		AND position_id IS NOT NULL
+		${includeShadows ? 'AND position_type_id < 2' : ''}
```
Comment (unknownunknowns1000) [inline lib/db/players.js:74]: I think this ternary is backwards, you want to filter if shadows are not included

Comment (lilserf) [reply lib/db/players.js:74]: I believe you're right

Comment (lilserf) [inline lib/db/players.js:74]: I'm bad at github

**Review by lilserf — APPROVED** [review]
I'll debug this and push a new version

**Pull request merged**